### DISCUSSION TITLE
네이버 예약 Refresh rate 설정 및실제 속도 문제

### DISCUSSION THREAD

**Discussion by hongluck** [kimytsc/covid-rest-vaccine-macro#46] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/52731362/128986824-b28320ee-3984-4092-84ea-de156b1c8696.png)

현재 delay는 1000으로 설정해놓은 상태입니다.

이미지 내의 업데이트시간 옆에 날짜와 업데이트 시도중이라는 글씨에는 꾸준히 업데이트를 진행하고 있지만

가운데 Bold 처리된 글씨의 refresh는 위에 시도중이라는 숫자를 못따라가는 느낌입니다.
Delay에 주어진 1초의 시간 보다 훨씬 길게 refresh되고 있는지요?

이유가 무엇인지 궁금합니다.

Bold 글씨의 업데이트는 거의 5초에 한번씩 이루어지고있는것같습니다
카카오같은경우 0.5초에 한번씩 잘 업데이트가되고있음을 확인했습니다.

**Comment by kimytsc** [comment, marked as answer]
회색으로 표시된 업데이트 시도 시간은 네이버 API를 호출한 시간입니다.

Bold로 표시되는 시간은 네이버의 DB가 갱신된 추정 시간으로, API에서 `vaccineLastSave`값으로 전달받은 시간을 그대로 노출하였습니다.

트래픽이 몰려 업데이트가 지연된다면, Bold로 표시되는 시간이 멈춰있을 것으로 판단됩니다.

**Reply by hongluck** [reply]
결국 bold값이 변동이 없어도 문제가 없다는 것인지요?
지금 500ms 로 계속하다보니 5분정도 멈춰있습니다

**Reply by kimytsc** [reply]
bold값이 변경이 없으면 API의 값을 못가져왔다는 의미가 되므로, 일시적인 차단이 되었을때 발생하는 현상입니다.

**Comment by kimytsc** [comment]
카카오의 경우, 네이버의 `vaccineLastSave`값과 같은 추정 시간이 없어 호출이 정상적으로 완료되었을때의 시간을 표시해두었습니다.

카카오 역시 트래픽이 몰려 업데이트가 지연되면, Bold로 표시되는 시간은 멈춰있습니다.
